### PR TITLE
LWM2M refactor firmware pull to make it reusable for other objects

### PR DESCRIPTION
**Pull request by DunderRoffe**
Refactored lwm2m_obj_firmware_pull.c so that the generic pull parts can be re-used.
I have tried my best to separate the commits to make the change easy to understand.

### CODE REVIEW

**Comment by DunderRoffe** [comment]
Dropped the Object 9 parts as this PR is complex enough without it

File: subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c [hunk 1]
```diff
@@ -18,7 +18,8 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include "lwm2m_pull_context.h"
 #include "lwm2m_engine.h"
 
-static void set_update_result_from_error(int error_code)
+static void set_update_result_from_error(struct firmware_pull_context *context,
+                                         int error_code)
 {
 	if (!error_code) {
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c:24]: `ARG_UNUSED(context);`

File: subsys/net/lib/lwm2m/lwm2m_pull_context.c [hunk 2]
```diff
@@ -186,6 +186,12 @@ do_firmware_transfer_reply_cb(const struct coap_packet *response,
 	uint8_t resp_code, *write_buf;
 	struct coap_block_context received_block_ctx;
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_pull_context.c:188]: No need for this empty line between declarations.

File: subsys/net/lib/lwm2m/lwm2m_pull_context.c [hunk 3]
```diff
@@ -278,7 +284,8 @@ do_firmware_transfer_reply_cb(const struct coap_packet *response,
 					goto error;
 				}
 
-				ret = write_cb(0, 0, 0,
+				ret = write_cb(context->obj_inst_id,
+								0, 0,
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_pull_context.c:288]: Alignment seems off here.

File: subsys/net/lib/lwm2m/lwm2m_pull_context.c [hunk 4]
```diff
@@ -315,6 +322,8 @@ static void do_transmit_timeout_cb(struct lwm2m_message *msg)
 {
 	int ret;
```
Comment (rlubos) [inline subsys/net/lib/lwm2m/lwm2m_pull_context.c:324]: No need for an empty line between declarations, rather add one after as checkpatch requires.

**Review by rlubos — COMMENTED** [review]
Thanks, not much to comment on since it looks pretty neat already. There are a few checkpatch issues though, please address them.

> Dropped the Object 9 parts as this PR is complex enough without it

Will you submit another PR once we get this in?

**Comment by DunderRoffe** [comment]
> Thanks, not much to comment on since it looks pretty neat already. There are a few checkpatch issues though, please address them.
Thx I will look into it

> Will you submit another PR once we get this in?

I am entering parental leave so I won't have any means of testing the code for a long while. I will ask someone else to do it though.

**Comment by DunderRoffe** [comment]
Sorry I am not going to have time to work on this for some time, life comes in the way.
Will try to reopen this when I get more time.

Pushing a dummy commit in the mean time as I do not want to stand for the sign off in the mean time.
Sorry for wasting your time :(

**Comment by rlubos** [comment]
No worries, let's close the PR for now then.

**Pull request closed without merging**